### ISSUE TITLE
Add cross-validation for ensemble weight optimization

### ISSUE THREAD

**Issue by datablogin** [datablogin/causal-inference-marketing#132]
## Problem
Currently, ensemble weights in G-computation are optimized on the same data used for training the individual models. This risks overfitting to the training data and may not generalize well to counterfactual predictions, which is the core use case in causal inference.

## Current Behavior
From `g_computation.py`:
```python
def _optimize_ensemble_weights(self, models, features, y):
    # Get predictions from each model  
    predictions = np.column_stack([models[name].predict(features) for name in model_names])
    
    # Optimize on same data
    def objective(weights):
        ensemble_pred = predictions @ weights
        mse = float(np.mean((y - ensemble_pred) ** 2))
        variance_penalty = self.ensemble_variance_penalty * float(np.var(weights))
        return mse + variance_penalty
```

## Proposed Solution
Implement cross-validation for ensemble weight optimization using one of these approaches:

### Option 1: K-Fold Cross-Validation
- Split data into K folds
- For each fold: train models, optimize weights on held-out fold
- Average weights across folds

### Option 2: Validation Split
- Use a fixed validation split (e.g., 20%)
- Train individual models on training set
- Optimize weights on validation set

### Option 3: Configurable Strategy
Add `ensemble_weight_strategy` parameter:
- `"in_sample"` - current behavior (fast but may overfit)
- `"cv"` - K-fold cross-validation (slower but better generalization)
- `"split"` - validation split (balanced)

## Implementation Considerations
1. **Computational cost**: CV adds K× overhead to weight optimization
2. **Sample size**: May not have enough data for CV in small samples
3. **Causal implications**: Need to ensure treatment balance across folds
4. **Bootstrap interaction**: How does this interact with bootstrap CIs?

## Related
- Referenced in PR #131 review (Major Issue #8)
- Related to ensemble overfitting concerns in causal inference

## Priority
Medium - Not blocking but important for production use with ensembles
